src/public/javascripts/batteries.js

```diff
@@ -1,225 +1,225 @@
-    }
-    if (i >= 56 && i <= 63) {
-      batteries[i] = new Battery(`Cell ${i}`, 7, (i - 56));
-      fillCell(i, 7, (i - 56));
-    }
-    if (i >= 64 && i <= 71) {
-      batteries[i] = new Battery(`Cell ${i}`, 8, (i - 64));
-      fillCell(i, 8, (i - 64));
-    }
-  }
-}
-
-// Run this function on window load
-window.onload = initBatteries;
-/**
- * Update table cell with battery cell info
- * @param {Number} batteryIndex index of battery
- */
-function updateCell(batteryIndex) {
-  let cell = batteries[batteryIndex];
-  let voltage = document.getElementById(`${cell.row}${cell.col}volt`);
-  voltage.innerHTML = `Voltage: ${cell.getVoltage()}`;
-}
-
-/**
- * Updates battery cell with voltage
- * @param {Number} batteryIndex Index of battery to update
- * @param {Number} voltage Voltage to put into battery
- */
-function updateBattery(batteryIndex, voltage) {
-  let cell = batteries[batteryIndex];
-  cell.setVoltage(voltage);
-}
-
-/**
- * Updates battery cell with voltage and renders it on the table
- * @param {Number} batteryIndex Index of battery to update
- * @param {Number} voltage Voltage to put into battery
- */
-function updateBatteryAndRender(batteryIndex, voltage) {
-  updateBattery(batteryIndex, voltage);
-  updateCell(batteryIndex);
-}
-
-/**
- * Updates the label at bottom of screen with temperatures
- * @param {Number} min Minimum temperature of battery pack
- * @param {Number} avg Average temperature of battery pack
- * @param {Number} max Maximum termperature of battery pack
- */
-function updateTemps(min, avg, max) {
-  document.getElementById('temps').innerHTML = `Min Cell Temp: ${min} C   Avg Cell Temp: ${avg} C   Max Cell Temp: ${max} C`;
-}
-
-/**
- * Checks if a cell's voltage is within the nominal values
- */
-function checkMinMax() {
-  let cell;
-  let tableCell;
-  let voltNum;
-  for (let i = 0; i < NUM_CELLS; i++) {
-    cell = batteries[i];
-    // console.log(cell);
-    tableCell = document.getElementById(`${cell.row}${cell.col}volt`).innerHTML;
-    voltNum = Number(tableCell.substr(9));
-    // console.log(`${i} ${tableCell}`);
-    if (voltNum <= MIN_VOLTAGE || voltNum >= MAX_VOLTAGE) {
-      document.getElementById(`${cell.row}${cell.col}cell`).style.backgroundColor = 'red';
-    } else {
-      document.getElementById(`${cell.row}${cell.col}cell`).style.backgroundColor = 'green';
-    }
-  }
-}
-// Check on given interval
-setInterval(checkMinMax, 1000);
-
-// On data in update and render voltages/ temps
-client.on('dataIn', (input) => {
-  console.log(input);
-  if (input.battery.cells) {
-    let cells = [];
-    for (let i = 0; i < input.battery.cells.length; i++) {
-      cells[i] = input.battery.cells[i];
-    }
-    for (let i = 0; i < cells.length; i++) {
-      console.log(`index ${i} volts ${cells[i]}`);
-      updateBatteryAndRender(i, cells[i]);
-    }
-    updateTemps(input.battery.minCellTemp, input.battery.avgCellTemp, input.battery.maxCellTemp);
-  }
-});
+/**

```

Comment:
If you are really riding the OOP train, I'd say consider making a table class a singleton class and keeping some of these variables as members again. Could make for easier reading.

Reply:
Redoing this whole page is on the to-do list. So when we rebuild it from scratch we can make a more OOP way of making the table.